src/state/swap/hooks.ts

```diff
@@ -74,7 +76,9 @@ export function tryParseAmount(value?: string, currency?: Currency): CurrencyAmo
     if (typedValueParsed !== '0') {
       return currency instanceof Token
         ? new TokenAmount(currency, JSBI.BigInt(typedValueParsed))
-        : CurrencyAmount.ether(JSBI.BigInt(typedValueParsed))
+        : chainId
+        ? CurrencyAmount.nativeCurrency(JSBI.BigInt(typedValueParsed), chainId)
```

Comment:
Is this two ternaries?

Reply:
Yes... it's not the best for clarity. I will rework it.

Reply:
13f1e0bece83146d489d1c53ae393f4de03d27ad